pymeasure/instruments/hcp/tc038d.py

```diff
@@ -55,89 +53,83 @@ class TC038D(Instrument):
+            data.append(elements * 2)  # 1B number of bytes to write
             for element in values:
-                for i in range(self.byteMode - 1, -1, -1):
-                    data.append(element >> i * 8 & 0xFF)
-        else:
-            raise TypeError(("Values has to be an integer or an iterable of "
-                             f"integers. values: {values}"))
+                data.extend(int(element).to_bytes(self.byteMode, byteorder="big", signed=True))
+        elif data[1] == self.functions['read']:
+            count = int(values[0]) * self.byteMode // 2 if values else self.byteMode // 2
+            data += [count >> 8, count & 0xFF]  # 2B number of elements to read
         data += CRC16(data)
         self.write_bytes(bytes(data))
+
+    def read(self):
+        """Read response and interpret the number"""
+        # Slave address, function
         got = self.read_bytes(2)
-        # slave address, function
-        if got[1] == self.functions['writeMultiple']:
+        if got[1] == self.functions['read']:
+            # length of data to follow
+            length = self.read_bytes(1)
+            # data length, 2 Byte CRC
+            read = self.read_bytes(length[0] + 2)
+            if read[-2:] != bytes(CRC16(got + length + read[:-2])):
+                raise ConnectionError("Response CRC does not match.")
+            return int.from_bytes(read[:-2], byteorder="big", signed=True)
+        elif got[1] == self.functions['writeMultiple']:
             # start address, number elements, CRC; each 2 Bytes long
             got += self.read_bytes(2 + 2 + 2)
             if got[-2:] != bytes(CRC16(got[:-2])):
                 raise ConnectionError("Response CRC does not match.")
-        else:
+        else:  # an error occurred
             end = self.read_bytes(3)  # error code and CRC
-            errors = {0x02: "Wrong start address",
-                      0x03: "Variable data error",
-                      0x04: "Operation error"}
-            raise ValueError(errors[end[0]])
-
-    @property
-    def setpoint(self):
-        """Get the current setpoint in °C."""
-        return int.from_bytes(self.readRegister(0x106), byteorder='big') / 10
-
-    @setpoint.setter
-    def setpoint(self, value):
-        """Set the setpoint in °C."""
-        value = int(round(value * 10, 0))
-        self.writeMultiple(0x106, value)
-
-    @property
-    def temperature(self):
-        """Get the current temperature in °C."""
-        return int.from_bytes(self.readRegister(0x0), byteorder='big') / 10
+            errors = {0x02: "Wrong start address.",
+                      0x03: "Variable data error.",
+                      0x04: "Operation error."}
+            if end[0] in errors.keys():
+                raise ValueError(errors[end[0]])
+            else:
+                raise ConnectionError(f"Unknown read error. Received: {got} {end}")
+
+    def check_errors(self):
+        """Read the response after setting a value."""
```

Comment:
Is the docstring aligned with purpose of method?

Reply:
The device always responds (even to setters), therefore I have to read afterwards.
The only way to do this right now, is to enable `check_set_errors` in the properties, which call check_errors.

So, I want to read, but have to call it 'check_errors'.

Reply:
Ok. I think, it could be explained in the docstring, since it's a bit cryptic.
BTW, in the past I implemented an instrument with similar behaviour and I think I overridden the `write` method.
I am a bit hesitant, but in SCPI instrument, you can call `check_errors` whenever you like, so it could mislead users.
No strong opinion on which way is better, feel free to choose what you think is more appropriate.

Reply:
I have the issue in the back of my mind, as I already proposed a small solution (allowing `control`'s setter to ask instead of write).
Maybe we can tackle it together with the split in Scpi an non Scpi instruments or together with the introduction of channels (with dynamic properties).